### DISCUSSION TITLE
Lookupdropdown seems to be absent from solution.

### DISCUSSION THREAD

**Discussion by mkardekar** [drivardxrm/LookupDropdown.PCF#36] [General]

**Comment by drivardxrm** [comment]
I'm not sure I understand your issue, can you elaborate or show a print screen?

**Reply by mkardekar** [reply]
Sure, thanks David for response.
I got the lookupdropdown however not sure if I'm doing it correctly since I'm consistently getting
"Loading" "Error Fetching data"
Scenario is something like this;
1) Entity A has a D365 form
2) Entity B has a single selection lookup dropdown on Entity A form.
3) Entity C is where PCF comes in for multiple selection based on selected value of field of Entity B.
Entity B has selected value has 1
then Entity C has 3 values 1a,1b, 1c
out of which I want to select 1a, 1b for multiple selection in Entity C.
Click on Save
4) When the same record is opened after sometime, I want all the selected values to auto populate for both single and multiple selected dropdowns.
Hope it's clear now.
Thanks
Nikhil

**Reply by drivardxrm** [reply]
As it is written in the documentation, you **must absolutely** configure the LookupDropdown PCF using the **classic** form editor. It will not work when configured from the modern from editor.  
see  this issue #29 
I have raised the issue to the product team but I have no ETA for a bug fix.

Tell me if it fixes your problem.

**Reply by mkardekar** [reply]
Hi David, I'm already using classic UI. I suspect it's a configuration issue. Hard to say what might be problem but for single selection it is all good.
Regards
Nikhil

**Reply by mkardekar** [reply]
[Screenshot.docx](https://github.com/drivardxrm/LookupDropdown.PCF/files/13608246/Screenshot.docx)

Screenshots attached.

**Reply by mkardekar** [reply]
Any luck?

**Reply by drivardxrm** [reply]
It's very hard for me to debug blindly.
if you make the error happen with the browsers Dev Tools opened (F12), do you see any specific error message ?

**Reply by mkardekar** [reply]
Hi David,

Please see the attached for error, can you pin point now on what could be the reason? I am guessing one or two but not sure.

Regards
Nikhil
[mac-copyofuat.crm6.dynamics.com-1702012397827.log](https://github.com/drivardxrm/LookupDropdown.PCF/files/13609176/mac-copyofuat.crm6.dynamics.com-1702012397827.log)

**Reply by mkardekar** [reply]
I was hoping to get it resolved over the weekend since the UI is part of POC which I want to showcase along with the advertisement of control which would help you in a way as well.

**Reply by drivardxrm** [reply]
I see 2 things.

1 - the control is complaining about :  
Cannot read properties of undefined (reading 'rp.mac_reportinggroup@OData.Community.Display.V1.FormattedValue') 

This seems to result in a malformed fetchxml query later on

2- 
```xml
<?xml version="1.0" encoding="UTF-8"?>
<fetch version="1.0" mapping="logical">
   <entity name="mac_focusarea">
      <filter type="and">
         <condition attribute="statecode" operator="eq" value="0" />
      </filter>
      <attribute name="mac_focusareaid" />
      <attribute name="mac_name" />
      <attribute name="undefined" />
      <link-entity xmlns="http://www.w3.org/1999/xhtml" name="mac_communitydevelopmentplan" from="mac_communitydevelopmentplanid" to="mac_cdplan" alias="dependent">
         <filter type="and">
            <condition attribute="mac_communitydevelopmentplanid" operator="eq" uitype="mac_communitydevelopmentplan" value="b29fb7a8-a153-ea11-a813-000d3a793530" />
         </filter>
      </link-entity>
   </entity>
</fetch>
```


As far as I can see, the relationship (filter) seems fine with the Dependent Lookup, see link entity

The problem comes from a field on the main entity that is not resolved correctly.
Have you tried to disable the Show record Image property?

![image](https://github.com/drivardxrm/LookupDropdown.PCF/assets/38399134/ddd0ddd7-0d97-46c4-a1ab-60fbbf24ebbf)

**Reply by mkardekar** [reply]
Thanks David, making record image as false does get rid of that error. 
However then it's not a multiselect dropdown which was expected, it is again single; so in fact lookup and dependent dropdown have same behaviour which is also OOB supported, nothing new so as to PCF. 
Perhaps for multiselect I need to check for other controls.

**Reply by mkardekar** [reply]
Any clues?

**Reply by drivardxrm** [reply]
This PCF control is not intended for MultiSelect dropdown indeed.
Since the backing field is of type Lookup.Simple, how would you map multiple selection to a single lookup value.

**Comment by mkardekar** [comment]
Irrelevant post and incorrect control discussion from me as well as previous comments were not looked in context by author or maintainer. Hence closing the discussion as inconclusive.
